### PR TITLE
sysrepo: Remove Lua bindings, build Python bindings in separate package

### PR DESCRIPTION
**Pull request by jefferyto**
Maintainer: @mislavn
Compile tested: armvirt-64, 2019-11-16 snapshot sdk
Run tested: armvirt-64, 2019-11-16 snapshot (imported module in Python only, no other functional tests)

Description:
This change:

* Removes building of Lua bindings. These bindings were [already removed][1] upstream.

* Fixes building of Python bindings:
  * Fix `SWIG_DIR` path (note that `SWIG_VERSION` will need to be updated whenever the swig package is upgraded)
  * Fix Python module paths and filenames

* Packages the Python bindings in a new package (`python3-sysrepo`) and builds it by default, instead of depending on a config option.

[1]: https://github.com/sysrepo/sysrepo/pull/1429

Signed-off-by: Jeffery To <jeffery.to@gmail.com>

### CODE REVIEW

File: net/sysrepo/Makefile [hunk 1]
```diff
@@ -21,25 +21,27 @@ PKG_LICENSE_FILES:=LICENSE
...
+  DEPENDS:=+libyang +libprotobuf-c +libev +libredblack +librt +libpthread
+endef
+
+define Package/python3-sysrepo
+  SECTION:=lang
+  CATEGORY:=Languages
+  SUBMENU:=Python
+  TITLE:=YANG-based data store library - Python 3 bindings
+  URL:=https://www.sysrepo.org/
+  DEPENDS:=+libsysrepo +libstdcpp +python3-base
```
Comment (neheb) [inline net/sysrepo/Makefile]: libstdc;pp should be implicitly included with libsysrepo.

**Comment by neheb** [comment]
I've no qualms with this. Other than this needing to be better maintained.

File: net/sysrepo/Makefile (hunk 1 above)
Comment (jefferyto) [reply net/sysrepo/Makefile]: If you mean that libstdcpp should already be a dependency of one of libsysrepo's dependencies, this doesn't appear to be the case. (I've installed this libsysrepo by itself and it doesn't pull in libstdcpp.)

If you mean the `+libstdcpp` should be moved from the python3-sysrepo `DEPENDS` to libsysrepo, I don't think libsysrepo has a real dependency on libstdcpp. The compiled module for the Python bindings (`_sysrepo.so`) depends on the C++ bindings (`libSysrepo-cpp.so`), and it's that C++ binding that depends on libstdcpp.

Comment (neheb) [reply net/sysrepo/Makefile]: Derp. I confused libyang with libyamp-cpp.

From looking at the code on github, it seems C++ is used in swig. I tested uClibc++. Unfortunately, it's using shared_ptr, which is not compatible.

**Pull request merged**